osbs/utils.py

```diff
@@ -341,6 +344,9 @@ def make_name_from_git(repo, branch, limit=53, separator='-'):
     while name.endswith('-'):
         name = name[:-1]
 
+    hash_str = b64encode(sha256(repo + branch).hexdigest())[:hash_size]
```

Comment:
It's more deterministic to compute the hash before repo and branch are truncated.